File: core/src/main/scala/com/xuxue/spider/core/scala/pluging/Pluging.scala
package com.xuxue.spider.core.scala.pluging

import com.xuxue.spider.core.scala.event.SpiderEvent
import com.xuxue.spider.core.scala.result.ResultIteam
import com.xuxue.spider.core.scala.task.Task
/**
  * Created by HanHan on 2016/7/8.
  */
trait Pluging[T] {

  val name:String

  val id:Int

  def filterTask(event:SpiderEvent[T]):Iterable[Task]

  def getResult(event:SpiderEvent[T]):ResultIteam

}

File: core/src/main/java/com/xuxue/spider/core/pluging/Pluging.java
package com.xuxue.spider.core.pluging;

import com.xuxue.spider.core.pipleline.ResultIteam;
import com.xuxue.spider.core.task.Task;

import java.util.LinkedList;


/**
 *
 */

public interface Pluging {

    LinkedList<Task> createChildTask(SpiderEvent e);

    LinkedList<ResultIteam> createResult(SpiderEvent e);

    String getName();

    int getID();
}
